src/app/date-picker/date-picker.directive.ts

```diff
@@ -0,0 +1,82 @@
+import {UtilsService} from '../common/services/utils/utils.service';
+import {OnChanges, SimpleChanges} from '@angular/core/core';
+import {ElementRef} from '@angular/core';
+import {IDatePickerConfig} from './date-picker-config.model';
+import {NgModel} from '@angular/forms';
+import {DatePickerComponent} from './date-picker.component';
+import {OnInit} from '@angular/core';
+import {ViewContainerRef} from '@angular/core';
+import { Directive, ComponentFactoryResolver, Input, HostListener } from '@angular/core';
+import { Moment } from 'moment';
+
+@Directive({ selector: '[dpDayPicker][ngModel]' })
+export class DatePickerDirective implements OnInit, OnChanges {
+  @Input('dpDayPicker') config: IDatePickerConfig = {
+    hideInputContainer: true,
+  };
+
+  @Input() inputElementContainer: ElementRef;
+  @Input() theme: string;
+
+  public datePicker: DatePickerComponent;
+
+  constructor(
+    public viewContainerRef: ViewContainerRef,
+    public componentFactoryResolver: ComponentFactoryResolver,
+    public model: NgModel,
+    public utils: UtilsService,
+  ) { }
+
+  ngOnInit(): void {
+    const factory = this.componentFactoryResolver.resolveComponentFactory(DatePickerComponent);
```

Comment:
pls move this logic to separate method , and it would be a good idea to break it to other methods. setting the elements, setting the validation and register listeners.